src/lib/authserver/ui/new_krb_inst.rb

```diff
@@ -140,102 +139,117 @@ def ok_handler
+    MITKerberos.append_to_log('%s' % out)
+    if !ok
+      Popup.Error(_('Failed to create password file! Log output may be found in /var/log/YaST/y2log'))
+      UI.ReplaceWidget(Id(:busy), Empty())
+      return
+    end
+    out, ok = MITKerberos.save_password_into_file(admin_dn, admin_pass, pass_file_path)
+    MITKerberos.append_to_log('%s' % out)
+    if !ok
+      Popup.Error(_('Failed to create password file! Log output may be found in /var/log/YaST/y2log'))
+      UI.ReplaceWidget(Id(:busy), Empty())
+      return
+    end
 
-      # Let kerberos do its initialisation sequence
-      out, ok = MITKerberos.init_dir(dir_addr, dm_dn, dm_pass, realm, container_dn, master_pass)
-      MITKerberos.append_to_log(out)
-      if !ok
-        Popup.Error(_('Kerberos initialisation failure! Log output may be found in %s') % [KDC_SETUP_LOG_PATH])
-        raise
-      end
+    # Let kerberos do its initialisation sequence
+    out, ok = MITKerberos.init_dir(dir_addr, dm_dn, dm_pass, realm, container_dn, master_pass)
+    MITKerberos.append_to_log('%s' % out)
+    if !ok
+      Popup.Error(_('Kerberos initialisation failure! Log output may be found in /var/log/YaST/y2log'))
+      UI.ReplaceWidget(Id(:busy), Empty())
+      return
+    end
 
-      # Kerberos may finally start
-      if !MITKerberos.restart_kdc
-        Popup.Error(_('Failed to start KDC, please inspect the journal of krb5kdc.service'))
-        raise
-      end
-      if !MITKerberos.restart_kadmind
-        Popup.Error(_('Failed to start kadmind, please inspect the journal of kadmind.service'))
-        raise
-      end
+    # Give kerberos rights to modify directory, relies on the kdc container existing
+    out, ok = ldap.aci_allow_modify(container_dn, 'kerberos-admin', admin_dn)
+    MITKerberos.append_to_log('%s' % out)
+    if !ok
+      Popup.Error(_('Failed to modify directory permission! Log output may be found in /var/log/YaST/y2log'))
+      UI.ReplaceWidget(Id(:busy), Empty())
+      return
+    end
+    out, ok = ldap.aci_allow_modify(container_dn, 'kerberos-kdc', kdc_dn)
+    MITKerberos.append_to_log('%s' % out)
+    if !ok
+      Popup.Error(_('Failed to modify directory permission! Log output may be found in /var/log/YaST/y2log'))
+      UI.ReplaceWidget(Id(:busy), Empty())
+      return
+    end
 
+    # Kerberos may finally start
+    if !MITKerberos.restart_kdc
+      Popup.Error(_('Failed to start KDC, please inspect the journal of krb5kdc.service'))
       UI.ReplaceWidget(Id(:busy), Empty())
-      Popup.Message(_('New instance has been set up! Log output may be found in %s') % [KDC_SETUP_LOG_PATH])
-      finish_dialog(:next)
-    rescue Exception => e
-      Popup.Error('There was an error ' + e.message)
-      # Give user an opportunity to correct mistake
+      return
+    end
+    if !MITKerberos.restart_kadmind
+      Popup.Error(_('Failed to start kadmind, please inspect the journal of kadmind.service'))
       UI.ReplaceWidget(Id(:busy), Empty())
```

Comment:
hint: this pattern is repeated many times, maybe it can be in own method. At least something like:
```
def report_error(msg)
  Popup.Error(msg)
  UI.ReplaceWidget(Id(:busy), Empty())
end

# and use it like
if !MITKerberos.restart_kadmind
  report_error(_('Failed to start kadmind, please inspect the journal of kadmind.service'))
  return
end
```

Reply:
I might do this in a future refactor?